oak_functions/loader/src/server.rs

```diff
@@ -680,13 +709,20 @@ impl WasmHandler {
         let mut wasm_state = self.init(request_bytes)?;
 
         wasm_state.invoke();
-        for mut extension in wasm_state
+        for extension in wasm_state
             .extensions_indices
             .take()
             .expect("no extensions_indices is set in wasm_state")
             .into_values()
         {
-            extension.terminate()?;
+            match extension {
+                Extension::Native(mut native_extension) => {
+                    native_extension.terminate()?;
+                }
+                Extension::Uwabi(_uwabi_extension) => {
+                    panic!("Uwabi Extension inserted in extension_indicies.")
```

Comment:
So why is it allowed? Is it possible to use the indexing only for Native extensions, and not Uwabi, using the type system?

Reply:
I think this part is only about termination logic. If UwabiExtensions don't have termination logic, perhaps an if-statement is more suitable here, than match and panic.

Reply:
Changed to `if let` statement so it is less confusing.